index.js

```diff
@@ -0,0 +1,53 @@
+const through2Concurrent = require('through2-concurrent');
+
+class MongoToSqs {
+
+  constructor(options) {
+    this.cursor = options.cursor; // db.collection(name).find()
+    this.sqs = options.sqs; // new AWS.SQS()
+    this.queueUrl = options.queueUrl; // https://sqs.us-east-1.amazonaws.com/000000000000/my-queue
+    this.formatPayload = options.formatPayload || (doc => doc);
+    this.concurrency = options.concurrency || 2500;
+    // TODO: this.sqsBatchSize - instead of sending every sqs doc individually
+  }
+
+  sendToQueue(payload) {
+    return this.sqs.sendMessage({
+      MessageBody: JSON.stringify(payload),
+      DelaySeconds: 0,
+      QueueUrl: this.queueUrl
+    }).promise();
+  }
+
+  start() {
+    let count = 0;
+    return new Promise((resolve, reject) => {
+      this.cursor.pipe(through2Concurrent.obj({
+        maxConcurrency: this.concurrency
+      }, (doc, encoding, done) => {
+        count += 1;
+        const payload = this.formatPayload(doc);
+        this.sendToQueue(payload)
+          .then(() => {
+            count -= 1;
+            done();
+          })
+          .catch(err => {
+            count -= 1;
+            console.log('sendToQueue:', doc, err);
+            done();
+          });
```

Comment:
How about using `.finally(done);` instead of repeating `done()` in success/reject blocks.

Reply:
aws-sdk defaults to native promises which doesn't support `.finally`

Reply:
That sucks :\